src/command/attributecommand.js

```diff
@@ -0,0 +1,201 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+import Command from './command.js';
+import TreeWalker from '../treemodel/treewalker.js';
+import Range from '../treemodel/range.js';
+import Selection from '../treemodel/selection.js';
+
+/**
+ * An extension of basic {@link core.command.Command} class, which provides utilities for a command that sets a single
+ * attribute on a text or element with value `true`. AttributeCommand uses {@link core.treeModel.Document#selection} to
+ * decide which nodes (if any) should be changed, and applies or removes attributes from them.
+ * See {@link core.treeView.Converter#execute} for more.
+ *
+ * The command checks {@link core.treeModel.Document#schema} to decide if it should be enabled.
+ * See {@link core.treeView.Converter#checkSchema} for more.
+ *
+ * @memberOf core.command
+ */
+export default class AttributeCommand extends Command {
```

Comment:
I'm looking at this class again and I feel that it will be crucial to make it (or related commands/classes) flexible and extensible. For instance, this command will need to share some logic with command which is able to set attribute with a specified value (e.g. the link command would need such a command). Then, we have base link feature extensions, like for instance link styles. I don't know yet how those extensions will be defined, either as separate commands or some logic pluggable into existing commands. Tough stuff to predict, so let's just keep this in our minds.